module/VuFind/src/VuFind/Db/AdapterFactory.php

```diff
@@ -106,12 +109,7 @@ public function __invoke(
      */
     public function getAdapter($overrideUser = null, $overridePass = null)
     {
-        // Parse details from connection string:
-        if (!isset($this->config->Database->database)) {
-            throw new \Exception('"database" setting missing');
-        }
-        return $this->getAdapterFromConnectionString(
-            $this->config->Database->database,
+        return $this->getAdapterFromConfig(
```

Comment:
Maybe I'm missing something, but it seems like the changes here break compatibility with the existing database configuration format. I'm definitely not opposed to introducing a more granular configuration format that's easier to parse and override (the current format is really a legacy from a different database library we used in version 1.0 more than 15 years ago), but we still need to include backward-compatibility for existing configuration files.

Reply:
I made a variation in the last commit